### PR TITLE
Test cb-init networkconfig plugin

### PR DESCRIPTION
**Pull request by cmin764**
- create specific scenario handling multiple networks
- test if the NIC(s) were properly configured
- fix bugs regarding cleanup

### CODE REVIEW

File: argus/tests/cloud/windows/test_smoke.py [hunk 1]
```diff
@@ -143,3 +144,52 @@ class TestCatchingSpecialize(base.TestBaseArgus):
...
+
+class TestStaticNetwork(base.TestBaseArgus):
+
+    def test_static_network(self):
+        """Check if the attached NICs were properly configured."""
+        # Get network adapter details within the windows instance.
+        instance_nics = self.introspection.get_nics()
+        # Filter them by DHCP disabled status for static checks.
+        instance_nics = filter(lambda nic: not nic["dhcp"], instance_nics)
+        # Retrieve and parse network details from the compute node.
```
Comment (cmin764) [inline argus/tests/cloud/windows/test_smoke.py]: Should I move this into the custom scenario/manager?

File: argus/introspection/cloud/windows.py [hunk 2]
```diff
@@ -240,3 +243,28 @@ def get_cloudconfig_executed_plugins(self):
             content = self.get_instance_file_content(path)
             files[basefile] = content.strip()
         return files
+
+    def get_nics(self):
```
Comment (PCManticore) [inline argus/introspection/cloud/windows.py]: You should provide a stub method for this on the base class, too.

File: argus/introspection/cloud/windows.py [hunk 3]
```diff
@@ -240,3 +243,28 @@ def get_cloudconfig_executed_plugins(self):
...
+        """Get IP available instance network adapters."""
+        # Retrieve utility script.
+        cmd = ("powershell Invoke-WebRequest -uri "
+               "{}/windows/network_details.ps1 -outfile "
+               "C:\\network_details.ps1".format(CONF.argus.resources))
+        self.remote_client.run_command_with_retry(cmd)
+        # Run and parse the output, where each adapter details
+        # block is separated by a new line.
+        cmd = "powershell C:\\network_details.ps1"
+        output = self.remote_client.run_command_verbose(cmd).strip()
```
Comment (PCManticore) [inline argus/introspection/cloud/windows.py:262]: Isn't calling this 'Get-WmiObject -ComputerName . Win32_NetworkAdapterConfiguration | Where-Object {$_.IPAddress -ne $null }' enough? Why do you have to echo the fields?

File: argus/scenarios/cloud.py [hunk 4]
```diff
@@ -48,6 +51,81 @@ def get_remote_client(self, username=None, password=None,
...
+        """Returns the network itself from the isolated network resources.
+
+        This works only with the isolated credentials and
+        this step is achieved by allowing/forcing tenant isolation.
+        """
+        # Extract the just created private network.
+        resources = self._isolated_creds.isolated_net_resources["primary"]
+        return resources[0]
+
+    def _fill_networks(self):
```
Comment (PCManticore) [inline argus/scenarios/cloud.py]: Just define a method, _get_networks, which returns the networks you desire, without setting self._networks to something.

File: argus/scenarios/cloud.py [hunk 5]
```diff
@@ -48,6 +51,81 @@ def get_remote_client(self, username=None, password=None,
...
+        """Explicitly gather and set the private networks.
+
+        All these networks will be attached to the newly created
+        instance without letting nova to handle this part.
+        """
+        networks = self._network_client.list_networks()["networks"]
+        _networks = []    # filtered and rearranged networks
```
Comment (PCManticore) [inline argus/scenarios/cloud.py]: Should be simpler and faster with a list comprehesion.

File: argus/tests/cloud/windows/test_smoke.py [hunk 6]
```diff
@@ -143,3 +144,52 @@ class TestCatchingSpecialize(base.TestBaseArgus):
...
+
+
+class TestStaticNetwork(base.TestBaseArgus):
+
+    def test_static_network(self):
+        """Check if the attached NICs were properly configured."""
+        # Get network adapter details within the windows instance.
+        instance_nics = self.introspection.get_nics()
+        # Filter them by DHCP disabled status for static checks.
+        instance_nics = filter(lambda nic: not nic["dhcp"], instance_nics)
```
Comment (PCManticore) [inline argus/tests/cloud/windows/test_smoke.py]: Use a list comprehestion. Plus, this won't work on Python 3 as you are using it right now.

File: argus/tests/cloud/windows/test_smoke.py (hunk 1 above)
Comment (PCManticore) [inline argus/tests/cloud/windows/test_smoke.py]: Move the part that retrieves the details of the host into a separate function.

**Comment by PCManticore** [comment]
Nice! There are some minor issues to fix before getting merged, but in general looks good.

File: argus/introspection/cloud/windows.py (hunk 3 above)
Comment (cmin764) [inline argus/introspection/cloud/windows.py:262]: I have to select what I need and print them in the order I choose. Also, there are other formatting and parsing purposes based on the simplicity of the values.

**Pull request merged**